### PR TITLE
Add terraform

### PR DESCRIPTION
**Pull request by maysakanoni**

### CODE REVIEW

File: .github/workflows/dotnet.yml [hunk 1]
```diff
@@ -99,8 +162,7 @@ jobs:
         id: deploy-to-webapp
         uses: azure/webapps-deploy@v2
         with:
-          app-name: 'housing-repairs-online-api-newark'
-          slot-name: 'Production'
+          app-name: "housing-repairs-online-api-newark"
```
Comment (jamiejordanmt) [inline .github/workflows/dotnet.yml:165]: Let's use the app service name secret here

**Review by jamiejordanmt — CHANGES_REQUESTED** [review]

File: .github/workflows/dotnet.yml [hunk 2]
```diff
@@ -99,8 +163,7 @@ jobs:
         id: deploy-to-webapp
         uses: azure/webapps-deploy@v2
         with:
-          app-name: 'housing-repairs-online-api-newark'
-          slot-name: 'Production'
+          app-name: "${{secrets.SERVICE_NAME}}-staging"
```
Comment (Bhavesh-Patel) [inline .github/workflows/dotnet.yml:166]: I don't think this should this have the `-staging` - is it required?

```suggestion
          app-name: "${{secrets.SERVICE_NAME}}"
```

File: terraform/app_service.tf [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    GOV_NOTIFY_KEY                        = var.gov_notify_key_production
+    INTERNAL_EMAIL                        = var.internal_email
+    INTERNAL_EMAIL_NOTIFY_TEMPLATE_ID     = var.internal_email_notify_template_id
+    JWT_SECRET                            = var.jwt_secret_production
+    SCHEDULING_API_URL                    = var.scheduling_api_url_production
+    SENTRY_DSN                            = var.sentry_dsn
+    SOR_CONFIGURATION                     = var.sor_configuration
+    STORAGE_CONTAINER_NAME                = var.storage_container_name_production
+
+  }
```
Comment (Bhavesh-Patel) [inline terraform/app_service.tf:41]: Please include the following to disable auth settings (which are enabled by default):
```
  auth_settings {
    enabled = false
  }
```

File: terraform/app_service.tf [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    GOV_NOTIFY_KEY                        = var.gov_notify_key_staging
+    INTERNAL_EMAIL                        = var.internal_email
+    INTERNAL_EMAIL_NOTIFY_TEMPLATE_ID     = var.internal_email_notify_template_id
+    JWT_SECRET                            = var.jwt_secret_staging
+    SCHEDULING_API_URL                    = var.scheduling_api_url_staging
+    SENTRY_DSN                            = var.sentry_dsn
+    SOR_CONFIGURATION                     = var.sor_configuration
+    STORAGE_CONTAINER_NAME                = var.storage_container_name_staging
+  }
+  site_config {}
```
Comment (Bhavesh-Patel) [inline terraform/app_service.tf:68]: Please include the following to disable auth settings (which are enabled by default):
```
  auth_settings {
    enabled = false
  }
```

**Review by Bhavesh-Patel — CHANGES_REQUESTED** [review]

File: .github/workflows/dotnet.yml (hunk 2 above)
Comment (maysakanoni) [reply .github/workflows/dotnet.yml:166]: Shouldn't this be the name of the app service slot? @Bhavesh-Patel

File: .github/workflows/dotnet.yml [hunk 5]
```diff
@@ -62,13 +62,76 @@ jobs:
...
           path: ${{env.DOTNET_ROOT}}/myapp
+  infrastructure:
+    name: "Provision Infrastructure"
+    env:
+      ARM_CLIENT_ID: ${{ secrets.AZURE_AD_CLIENT_ID }}
+      ARM_CLIENT_SECRET: ${{ secrets.AZURE_AD_CLIENT_SECRET }}
+      ARM_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+      ARM_TENANT_ID: ${{ secrets.AZURE_AD_TENANT_ID }}
+    runs-on: ubuntu-latest
+    environment: "Staging"
```
Comment (Bhavesh-Patel) [inline .github/workflows/dotnet.yml:73]: This specifies the GitHub actions environment, and `Staging` doesn't have anything specific defined within it.

It is required?

**Review by Bhavesh-Patel — DISMISSED** [review]

**Review by jamiejordanmt — APPROVED** [review]

**Review by jamiejordanmt — APPROVED** [review]

**Pull request merged**